source/extensions/geoip_providers/maxmind/geoip_provider.h

```diff
@@ -97,16 +103,29 @@ class GeoipProvider : public Envoy::Geolocation::Driver,
   // Allow the unit test to have access to private members.
   friend class GeoipProviderPeer;
   GeoipProviderConfigSharedPtr config_;
-  MaxmindDbPtr city_db_;
-  MaxmindDbPtr isp_db_;
-  MaxmindDbPtr anon_db_;
-  MaxmindDbPtr initMaxMindDb(const absl::optional<std::string>& db_path);
+  // These three mutexes are never held at the same time. We signify this by requiring
+  // that all three are 'acquired before' the others, since there is no exclusion annotation.
```

Comment:
If they're never acquired at the same time, why not just use one mutex?

Reply:
answered below